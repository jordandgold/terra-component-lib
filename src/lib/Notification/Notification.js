import React from "react";
import "./Notification.scss";
import Icon from "../Icon/Icon";
import PropTypes from "prop-types";

const Notification = props => {
  const { onClick, text, type } = props;
  return (
    <section className={`ter-notification ter-notification--${type}`}>
      <button
        className="ter-notification__close"
        onClick={onClick}
        aria-label="Close Notification"
      >
        <Icon type="enclosed-x-dark-16px" size="16px" />
      </button>
      <div className="ter-notification__body">
        <p>{text}</p>
      </div>
    </section>
  );
};

export default Notification;

Notification.propTypes = {
  text: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
};
