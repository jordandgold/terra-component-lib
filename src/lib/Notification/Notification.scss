@import "../global-styles/terra.scss";

@mixin notification-base() {
    font-size: font-size('sm');
    background-color: $noficiation-bg-color;
    padding: spacing('xs') spacing('sm');
    display: block;
    position: relative;
    border-radius: $notification-border-radius;
    margin-bottom: spacing('sm');
    &__close {
        position: absolute;
        top: 1rem;
        right: 1rem;
        opacity: 0.4;
        padding: 0;
        margin: 0;
        background: transparent;
        border: 0;
        &:after {
            @include icon-base();
			content: '';
			position: absolute;
			top: 50%;
			right: 0px;
			transform: rotate(0deg) translateY(-50%);
			transition: transform 200ms ease-in-out;
			background-position: -316px -316px;
            height: 20px;
            width: 20px;
		}
        &:hover {
            opacity: 0.6;
            cursor: pointer;
        }
    }
    &__body {
        p:last-child {
            margin-bottom: 0;
        }
    }
}

@mixin notification-variant($color, $bg: transparent) {
    border-bottom: 3px solid darken($color, 10%);
    [class$="close"] {
        i {
            color: neutral-color('white');
        }
    }
    hr {
        opacity: 0.4;
        background-color: darken($color, 10%);
    }
    a {
        color: lighten($color, 20%);
        text-decoration: underline;
        &:hover {
            color: lighten($color, 10%);
            text-decoration: underline;
        }
    }
}

.ter-notification {
    @include notification-base();
}

@each $color, $value in $emotive-colors {
  .ter-notification--#{$color} {
    @include notification-variant($value);
  }
}
