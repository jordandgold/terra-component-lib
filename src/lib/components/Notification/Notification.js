import React, { Component } from "react";
import "./Notification.scss";
import PropTypes from "prop-types";

class Notification extends Component {
  render() {
    const { onClick, text } = this.props;
    return (
      <section className="notification">
        <p>{text}</p>
        <button className="notification-close-icon" onClick={onClick} />
      </section>
    );
  }
}

export default Notification;

Notification.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
};
