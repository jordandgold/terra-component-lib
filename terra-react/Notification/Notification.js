import React from "react";
import "./Notification.scss";
import Icon from "../Icon/Icon";

var Notification = function Notification(props) {
  var onClick = props.onClick,
      text = props.text,
      type = props.type;
  return React.createElement("section", {
    className: "ter-notification ter-notification--".concat(type)
  }, React.createElement("button", {
    className: "ter-notification__close",
    onClick: onClick,
    "aria-label": "Close Notification"
  }, React.createElement(Icon, {
    type: "enclosed-x-dark-16px",
    size: "16px"
  })), React.createElement("div", {
    className: "ter-notification__body"
  }, React.createElement("p", null, text)));
};

export default Notification;