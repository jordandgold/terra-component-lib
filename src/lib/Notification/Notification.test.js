import React from "react";
import { shallow } from "enzyme";

import Notification from "./Notification";

describe("Notification", () => {
  let wrapper;
  let mockOnClick;

  beforeEach(() => {
    mockOnClick = jest.fn();
    const mockText = "This is a mock notification message";
    const mockType = "default";
    wrapper = shallow(<Notification onClick={mockOnClick} text={mockText} type={mockType} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call onClick when clicked", () => {
    wrapper.find("button").simulate("click");

    expect(mockOnClick).toHaveBeenCalled();
  });
});
