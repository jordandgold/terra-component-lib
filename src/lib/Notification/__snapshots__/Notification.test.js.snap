// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should match the snapshot 1`] = `
<section
  className="ter-notification ter-notification--default"
>
  <button
    aria-label="Close Notification"
    className="ter-notification__close"
    onClick={[MockFunction]}
  >
    <Icon
      size="16px"
      type="enclosed-x-dark-16px"
    />
  </button>
  <div
    class="ter-notification__body"
  >
    <p>
      This is a mock notification message
    </p>
  </div>
</section>
`;
